sonaranalyzer-dotnet/rspec/cs/S2589_c#.html

```diff
@@ -2,38 +2,51 @@
 because it does not match the programmer's intent, then it's a bug and the expression should be fixed.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
-a = true;
-if (a) // Noncompliant
+public void Sample(bool b, bool c, string s)
 {
-  DoSomething();
-}
+    var a = true;
+    if (a) // Noncompliant
+    {
+        DoSomething();
+    }
 
-if (b &amp;&amp; a) // Noncompliant; "a" is always "true"
-{
-  DoSomething();
-}
+    if (b &amp;&amp; a) // Noncompliant; "a" is always "true"
+    {
+        DoSomething();
+    }
 
-if (c || !a) // Noncompliant; "!a" is always "false"
-{
-  DoSomething();
+    if (c || !a) // Noncompliant; "!a" is always "false"
+    {
+        DoSomething();
+    }
+
+    string d = null;
+    var v1 = d ?? "value"; // Noncompliant, d is always null
+    var v2 = s ?? d; // Noncompliant, d is always null and produced value is always equal to s. Condition to check value of s is gratuitous.
```

Comment:
should use the definite article "the produced value"

Reply:
"The condition to check the value of s is gratuitous." or "Checking the value of s is gratuitous."

Reply:
RSPEC is updated. Files will be updated as part of release process.

Reply:
that was quick :)